Fix getting the executable path from the `mcp setup claude` command

pepicrft opened the pull request:
### Short description 📝
`CommandLine.arguments[0]` doesn't yield the absolute path of the executable. For example if you run it through `mise` it resolves to just `tuist`. I adjusted it to use a Darwing-specific API that returns the whole absolute path.

### How to test the changes locally 🧐

1. Build `tuist`
2. `cd` into the derived data directory containing `tuist`
3. Run `./tuist mcp setup claude`
4. You should get the absolute path in `~/Library/Application\ Support/Claude/claude_desktop_config.json`

### Contributor checklist ✅

- [x] The code has been linted using run `mise run lint-fix`
- [x] The change is tested via unit testing or acceptance testing, or both
- [x] The title of the PR is formulated in a way that is usable as a changelog entry
- [ ] In case the PR introduces changes that affect users, the documentation has been updated

### Reviewer checklist ✅

- [ ] The code architecture and patterns are consistent with the rest of the codebase
- [ ] Reviewer has checked that, if needed, the PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`, and the title is usable as a changelog entry

fortmarek reviewed (commented):
When I was doing the build insights, I added [this](https://github.com/tuist/tuist/blob/b935374a74840c9ba655ed36d52c17cce518f318/Sources/TuistSupport/Utils/Environment.swift#L216-L218) property that does resolve to the full path even when using mise. Can you try that instead?

pepicrft commented on Sources/TuistKit/MCP/MCPServerCommandResolver.swift:
```diff
@@ -9,11 +10,21 @@ protocol MCPServerCommandResolving {
         return (executablePath, ["mcp", "start"])
     }
+
+    private static func getExecutablePath() -> String! {
```
I think it's safe to assume here that we'll get a value.

pepicrft commented:
> When I was doing the build insights, I added [this](https://github.com/tuist/tuist/blob/b935374a74840c9ba655ed36d52c17cce518f318/Sources/TuistSupport/Utils/Environment.swift#L216-L218) property that does resolve to the full path even when using mise. Can you try that instead?

That version only covers the scenario where the executable is a relative path, not the one where the name of the executable, (e.g. `tuist`) is resolved by the system using the `PATH` environment variable. I adjusted that getter in `Environment` to apply the fix to both.

fortmarek commented:
> That version only covers the scenario where the executable is a relative path, not the one where the name of the executable, (e.g. tuist) is resolved by the system using the PATH environment variable. I adjusted that getter in Environment to apply the fix to both.

Got it, that was the piece I was missing. Then we can go with the changes here.
